mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -5,821 +5,745 @@
+    io::Cursor,
+    time::{SystemTime, UNIX_EPOCH},
 };
 
-use bincode::{
-    deserialize_from,
-    serialize,
-    serialize_into,
-    serialized_size,
-    Error,
-};
+use bincode::{deserialize_from, serialize, serialize_into, Error};
+
 use crc::{crc32, Hasher32};
 use serde::{Deserialize, Serialize};
 use snafu::{ResultExt, Snafu};
 
 use crate::{
-    bdev::nexus::{
-        nexus_bdev::Nexus,
-        nexus_child::{ChildError, NexusChild},
-        nexus_label::{Aligned, GptEntry, GptGuid, LabelError},
-        nexus_metadata_content::NexusConfig,
-    },
+    bdev::nexus::{nexus_child::NexusChild, nexus_label::GptGuid as Guid},
     core::{CoreError, DmaBuf, DmaError},
 };
 
 #[derive(Debug, Snafu)]
 pub enum MetaDataError {
-    #[snafu(display("{}", source))]
-    NexusChildError { source: ChildError },
-    #[snafu(display("Error probing disk label: {}", source))]
-    ProbeLabelError { source: LabelError },
-    #[snafu(display("Error reading {}: {}", name, source))]
-    ReadError { name: String, source: CoreError },
-    #[snafu(display("Error writing {}: {}", name, source))]
-    WriteError { name: String, source: CoreError },
+    #[snafu(display("Serialization error: {}", source))]
+    SerializeError { source: Error },
+    #[snafu(display("Deserialization error: {}", source))]
+    DeserializeError { source: Error },
     #[snafu(display(
         "Failed to allocate buffer for reading {}: {}",
         name,
         source
     ))]
-    ReadAlloc { name: String, source: DmaError },
+    ReadAlloc { source: DmaError, name: String },
     #[snafu(display(
         "Failed to allocate buffer for writing {}: {}",
         name,
         source
     ))]
-    WriteAlloc { name: String, source: DmaError },
-    #[snafu(display("Serialization error: {}", source))]
-    SerializeError { source: Error },
-    #[snafu(display("Deserialization error: {}", source))]
-    DeserializeError { source: Error },
+    WriteAlloc { source: DmaError, name: String },
+    #[snafu(display("Error reading {}: {}", name, source))]
+    ReadError { source: CoreError, name: String },
+    #[snafu(display("Error writing {}: {}", name, source))]
+    WriteError { source: CoreError, name: String },
     #[snafu(display(
-        "Incorrect MetaData header size: actual={} expected={}",
-        actual_size,
-        expected_size
+        "Failed to obtain BdevHandle for child {}: {}",
+        name,
+        source
     ))]
-    HeaderSize {
-        actual_size: u32,
-        expected_size: u32,
-    },
-    #[snafu(display("Incorrect MetaData header signature"))]
-    HeaderSignature {},
-    #[snafu(display("Incorrect MetaData header checksum"))]
-    HeaderChecksum {},
-    #[snafu(display("Incorrect MetaData index checksum"))]
-    IndexChecksum {},
-    #[snafu(display("Incorrect MetaData configuration object checksum"))]
+    HandleError { source: CoreError, name: String },
+    #[snafu(display("Incorrect MetaDataObject signature"))]
```

Comment:
what it be useful ti display what all of these failed things are, what what the correct ones should have been?

Reply:
In my view the actual values of these signatures/checksums aren't all that useful. They are just to ensure that the bytes read from the MayaMeta partition actually represent a valid object